src/OpenTelemetry.Exporter.OneCollector/Internal/CallbackManager.cs

```diff
@@ -33,10 +33,14 @@ public IDisposable Add(T callback)
 
         lock (this.lockObject)
         {
+#if NET7_0_OR_GREATER
+            ObjectDisposedException.ThrowIf(this.disposed, nameof(CallbackManager<T>));
+#else
```

Comment:
For now we have only one place, so it is fine. Consider to extend Guard class with implementation for `ThrowIfObjectDisposedExcpetion(bool disposed, Type type)`